tests/testthat/test-implicit_assignment_linter.R

```diff
@@ -162,23 +162,18 @@ test_that("implicit_assignment_linter respects except argument", {
   )
 })
 
-test_that("implicit_assignment_linter makes exceptions for functions that capture side-effects", {
-  linter <- implicit_assignment_linter()
+test_that("implicit_assignment_linter skips allowed usages with braces", {
+  linter <- implicit_assignment_linter(except = character(0L))
 
-  # base
-  expect_lint("output <- capture.output(x <- f())", NULL, linter)
-  expect_lint("quote(a <- 1L)", NULL, linter)
-  expect_lint("bquote(a <- 1L)", NULL, linter)
-  expect_lint("expression(a <- 1L)", NULL, linter)
+  expect_lint("output <- capture.output({ x <- f() })", NULL, linter)
```

Comment:
looks like these are all side-effect functions... my recollection (CMIIW) was we wanted to allow `<-` within braces no matter the parent function.

If that's the case, could you make some of these tests use a generic function like `foo({ a <- 1L })`?

It may also be preferable to keep the code aligned to the rest of the default linters, which means using

```r
foo({
  a <- 1L
})
```

because of `brace_linter()`.